compare_graph.py

```diff
@@ -0,0 +1,210 @@
+    network_number_of_edges = graph.number_of_edges()
+    network_number_of_nodes = graph.number_of_nodes()
+
+    # Create a dictionary to store the extracted network properties
+    network_properties = {
+        "network_edges": network_edges,
+        "network_degree": network_degree,
+        "network_average_degree_connectivity": network_average_degree_connectivity,
+        "network_nodes": network_nodes,
+        "network_number_of_edges": network_number_of_edges,
+        "network_number_of_nodes": network_number_of_nodes,
+    }
+
+    return network_properties
+
+
+def plot_network_graph(network: pypsa.Network, seed: int = 1969) -> None:
+    """
+    Plots the network graph.
+
+    Args:
+        network: pypsa.Network
+    Returns:
+        None
+    """
+    graph = network.graph()
+    options = {"node_color": "black", "node_size": 50,
+               "linewidths": 0, "width": 0.1}
+    pos = nx.spring_layout(graph, seed=seed)  # Seed for reproducible layout
+    nx.draw(graph, pos, **options)
+    plt.savefig("network_graph.png", dpi=300)
+
+
+def available_item_properties(
+    network1: pypsa.Network, network2: pypsa.Network, item: str
+) -> list:
+    """
+    Identifies items present in both network1 and network2 for the specified item type.
+
+    Args:
+        network1 (pypsa.Network): The first network to compare.
+        network2 (pypsa.Network): The second network to compare.
+        item (str): The type of item to check for availability, either "network_nodes" or "network_edges".
+
+    Returns:
+        list: A list of items present in both network1 and network2.
+    """
+
+    if item not in ["network_nodes", "network_edges"]:  # Validate item type
+        raise ValueError(
+            "Invalid item type. Must be 'network_nodes' or 'network_edges'."
+        )
+
+    available_item = []  # Initialize a list to store common items
+
+    # Iterate through items in network1 and check for presence in network2
+    for idx in network1[item]:
+        if idx in network2[item]:
+            available_item.append(idx)
+
+    return available_item
+
+
+def missing_item_properties(
+    network1: pypsa.Network, network2: pypsa.Network, item: str
+) -> list:
+    """
+    Identifies items present in network1 but missing in network2 for the specified item type.
+
+    Args:
+        network1 (pypsa.Network): The first network to compare.
+        network2 (pypsa.Network): The second network to compare.
+        item (str): The type of item to check for missing entries, either "network_nodes" or "network_edges".
+
+    Returns:
+        list: A list of items present in network1 but missing in network2.
+    """
+
+    if item not in ["network_nodes", "network_edges"]:  # Validate item type
+        raise ValueError(
+            "Invalid item type. Must be 'network_nodes' or 'network_edges'."
+        )
+
+    missing_item = []  # Initialize a list to store missing items
```

Comment:
unfortunately this won't do what we want it to - here you want to compare the graphs without caring about the details of the buses making up the graph. Looking at the function graph.add_nodes_from(buses_i) at Line 64 of pypsa/graph.py, the function you've got here for comparing nodes would actually compare the lists of buses in each network. What we want to do instead is to compare e.g. the number of nodes in each graph

Reply:
Actually, the pypsa-network is translated into a graph by using the buses as nodes and the links/links as edges. Treating the network as a graph means that no attention is paid to the size of the bus or the length of the links/lines.

Feel free to disagree :)

Reply:
Also, the graph method in pypsa was built ontop of `networkx` which is used to develop graphs in Python https://github.com/PyPSA/PyPSA/blob/master/pypsa/graph.py#L22

Reply:
[here](https://github.com/PyPSA/PyPSA/blob/4298c23ed625fa3e0e4cecfb84335b27b76e2ac6/pypsa/graph.py#L64) in the pypsa classs for graphs the graph is created from the set of buses:
`graph.add_nodes_from(buses_i)`
At [line 53](https://github.com/PyPSA/PyPSA/blob/4298c23ed625fa3e0e4cecfb84335b27b76e2ac6/pypsa/graph.py#L53) of the same file, we set 
`buses_i = network.buses.index`
As such, the nodes in the graph each correspond to the string index of the associated bus. This means that two nodes in the graph are defined to be equal if they have the same bus name. This means that when in your method you compare the nodes, you are only comparing the respective bus names in each network. This is not the information we want. Instead, we should look at general properties of the graph e.g. number of nodes, average degree.

Reply:
I could remove the nodes and edges properties from the code. But there can't be multiple buses with the same name in `network.buses.index`. I only added in case we need to do some more complex comparison which can also be useful for our matching algorithm which also relies on the index of buses.

let me know what you think?

Reply:
the problem is that it isn't the names of the buses that we are interested in - it is their other properties